src/actions/searchActions.js

```diff
@@ -0,0 +1,24 @@
+import { instance } from '../utils/axios';
+import handleMessages from '../utils/messages';
+import actionTypes from './types';
+
+const searchActions = (filter, phrase, history) => async (dispatch) => {
+  if (!phrase) {
```

Comment:
It is becoming too much interactive. You can remove this `if else` statement and add this . `handleMessages('loading', 'Searching...');` in the try block.

Reply:
Done